### PR TITLE
Some fixes, mostly to cmake commands

### PR DESCRIPTION
**Pull request by mathieu-aubin**
- -s to default cflags/cxxflags
- cmake generates Unix Makefiles everywhere
- cmake verbose makefile
- aom: disable docs
- opus: adds --disable-doc
- rabbitmq: disables all docs and tests in cmake command
- rubberband: fix -Dno_shared=true to -Ddefault_library=static and add resampler
- mysofa: build tests to off
- davs2: enable-strip to configure args

maybe some other things here and there

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -50,8 +46,8 @@ RUN apk add --no-cache \
 # -static-libgcc is needed to make gcc not include gcc_s as "as-needed" shared library which
 # cmake will include as a implicit library.
 # other options to get hardened build (same as ffmpeg hardened)
-ARG CFLAGS="-O3 -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC"
-ARG CXXFLAGS="-O3 -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC"
+ARG CFLAGS="-O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG"
+ARG CXXFLAGS="-O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG"
```
Comment (wader) [inline Dockerfile:50]: Is the idea that `-DNDEBUG` disables asserts? personally i usually like that assert aborts even in release builds, crash early :)

File: Dockerfile [hunk 2]
```diff
@@ -223,7 +237,7 @@ RUN \
   wget -O lame.tar.gz "$MP3LAME_URL" && \
   echo "$MP3LAME_SHA256  lame.tar.gz" | sha256sum --status -c - && \
   tar xf lame.tar.gz && \
-  cd lame-* && ./configure --disable-shared --enable-static --enable-nasm --disable-gtktest --disable-cpml --disable-frontend && \
+  cd lame-* && ./configure --disable-shared --enable-static --enable-nasm --disable-gtktest --disable-cpml --disable-frontend --disable-rpath && \
```
Comment (wader) [inline Dockerfile:240]: Will rpath stuff affect static PIE builds? thinking rpath is usually a dynamic lib runtime linking thingy or?

File: Dockerfile [hunk 3]
```diff
@@ -86,7 +82,19 @@ RUN \
   git clone --depth 1 --branch v$AOM_VERSION "$AOM_URL" && \
   cd aom && test $(git rev-parse HEAD) = $AOM_COMMIT && \
   mkdir build_tmp && cd build_tmp && \
-  cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DENABLE_EXAMPLES=NO -DENABLE_TESTS=NO -DENABLE_TOOLS=NO -DCONFIG_TUNE_VMAF=1 -DENABLE_NASM=ON -DCMAKE_INSTALL_LIBDIR=lib .. && \
+  cmake \
+    -G"Unix Makefiles" \
+    -DCMAKE_VERBOSE_MAKEFILE=ON \
```
Comment (wader) [inline Dockerfile:87]: Does this make it more verbose than the default verbosity?

File: Dockerfile [hunk 4]
```diff
@@ -86,7 +82,19 @@ RUN \
   git clone --depth 1 --branch v$AOM_VERSION "$AOM_URL" && \
   cd aom && test $(git rev-parse HEAD) = $AOM_COMMIT && \
   mkdir build_tmp && cd build_tmp && \
-  cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DENABLE_EXAMPLES=NO -DENABLE_TESTS=NO -DENABLE_TOOLS=NO -DCONFIG_TUNE_VMAF=1 -DENABLE_NASM=ON -DCMAKE_INSTALL_LIBDIR=lib .. && \
+  cmake \
+    -G"Unix Makefiles" \
```
Comment (wader) [inline Dockerfile:86]: Wonder if makefile generator is default on unix systems?

**Comment by wader** [comment]
Nice cleanup! like it. Added some comment

File: Dockerfile (hunk 4 above)
Comment (wader) [reply Dockerfile:86]: Yeah guess so too, but maybe make sense to make it explicit if we use `-DCMAKE_*` defines? have no strong opinions about it

File: Dockerfile (hunk 3 above)
Comment (mathieu-aubin) [reply Dockerfile:87]: From [cmake documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_VERBOSE_MAKEFILE.html)

**Comment by wader** [comment]
Wanted to see the difference between CMAKE_VERBOSE_MAKEFILE=ON and CMAKE_VERBOSE_MAKEFILE=OFF. It quite nice to quickly be able to see that we use expected flags
```
#6 19.25 -- Build files have been written to: /aom/build_tmp
#6 19.36 [  1%] Building CXX object CMakeFiles/yuv.dir/third_party/libyuv/source/convert_argb.cc.o
#6 19.37 [  1%] Updating version info if necessary.
#6 19.43 [  1%] Built target aom_version_check
#6 19.47 [  1%] Building C object CMakeFiles/aom_version.dir/gen_src/aom_version_dummy.c.o
#6 19.52 [  1%] Linking C static library libaom_version.a
#6 19.56 [  2%] Built target aom_version
#6 19.57 [  3%] Writing aom.pc
#6 19.63 [  3%] Building C object CMakeFiles/aom_pc.dir/gen_src/aom_pc_dummy.c.o
#6 19.71 [  3%] Linking C static library libaom_pc.a
#6 19.76 [  3%] Built target aom_pc
#6 19.78 [  3%] Building C object CMakeFiles/aom_rtcd.dir/aom_dsp/aom_dsp_rtcd.c.o
#6 22.55 [  3%] Building CXX object CMakeFiles/yuv.dir/third_party/libyuv/source/cpu_id.cc.o
#6 22.96 [  3%] Building CXX object CMakeFiles/yuv.dir/third_party/libyuv/source/planar_functions.cc.o
#6 24.43 [  3%] Building C object CMakeFiles/aom_rtcd.dir/aom_scale/aom_scale_rtcd.c.o
#6 24.59 [  4%] Building C object CMakeFiles/aom_rtcd.dir/av1/common/av1_rtcd.c.o
#6 25.34 [  5%] Built target aom_rtcd
#6 25.41 [  5%] Building C object CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_quantize_avx2.c.o
#6 26.24 [  5%] Building CXX object CMakeFiles/yuv.dir/third_party/libyuv/source/row_any.cc.o
#6 27.07 [  5%] Building C object CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_highbd_quantize_avx2.c.o
#6 28.47 [  6%] Building C object CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/corner_match_avx2.c.o
#6 29.81 [  6%] Building C object CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/error_intrin_avx2.c.o
#6 31.10 [  6%] Building C object CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/highbd_block_error_intrin_avx2.c.o
#6 32.20 [  7%] Building C object CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_fwd_txfm2d_avx2.c.o
#6 32.85 [  8%] Building CXX object CMakeFiles/yuv.dir/third_party/libyuv/source/row_common.cc.o
^C
```

```
#6 16.39 -- Build files have been written to: /aom/build_tmp
#6 16.44 /usr/bin/cmake -S/aom -B/aom/build_tmp --check-build-system CMakeFiles/Makefile.cmake 0
#6 16.46 /usr/bin/cmake -E cmake_progress_start /aom/build_tmp/CMakeFiles /aom/build_tmp//CMakeFiles/progress.marks
#6 16.47 make  -f CMakeFiles/Makefile2 all
#6 16.48 make[1]: Entering directory '/aom/build_tmp'
#6 16.48 make  -f CMakeFiles/aom_version_check.dir/build.make CMakeFiles/aom_version_check.dir/depend
#6 16.52 make  -f CMakeFiles/yuv.dir/build.make CMakeFiles/yuv.dir/depend
#6 16.52 make[2]: Entering directory '/aom/build_tmp'
#6 16.52 cd /aom/build_tmp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /aom /aom /aom/build_tmp /aom/build_tmp /aom/build_tmp/CMakeFiles/aom_version_check.dir/DependInfo.cmake --color=
#6 16.55 make[2]: Leaving directory '/aom/build_tmp'
#6 16.55 make  -f CMakeFiles/aom_version_check.dir/build.make CMakeFiles/aom_version_check.dir/build
#6 16.57 make[2]: Entering directory '/aom/build_tmp'
#6 16.57 make[2]: Entering directory '/aom/build_tmp'
#6 16.57 cd /aom/build_tmp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /aom /aom /aom/build_tmp /aom/build_tmp /aom/build_tmp/CMakeFiles/yuv.dir/DependInfo.cmake --color=
#6 16.58 [  0%] Updating version info if necessary.
#6 16.59 /usr/bin/cmake -DAOM_CONFIG_DIR=/aom/build_tmp -DAOM_ROOT=/aom -DGIT_EXECUTABLE=/usr/bin/git -DPERL_EXECUTABLE=/usr/bin/perl -P /aom/build/cmake/version.cmake
#6 16.63 make[2]: Leaving directory '/aom/build_tmp'
#6 16.63 make  -f CMakeFiles/yuv.dir/build.make CMakeFiles/yuv.dir/build
#6 16.63 make[2]: Entering directory '/aom/build_tmp'
#6 16.64 make[2]: Leaving directory '/aom/build_tmp'
#6 16.65 [  1%] Building CXX object CMakeFiles/yuv.dir/third_party/libyuv/source/convert_argb.cc.o
#6 16.65 /usr/bin/c++  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c++11 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=240000 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wno-unused-parameter -MD -MT CMakeFiles/yuv.dir/third_party/libyuv/source/convert_argb.cc.o -MF CMakeFiles/yuv.dir/third_party/libyuv/source/convert_argb.cc.o.d -o CMakeFiles/yuv.dir/third_party/libyuv/source/convert_argb.cc.o -c /aom/third_party/libyuv/source/convert_argb.cc
#6 16.65 [  1%] Built target aom_version_check
#6 16.65 make  -f CMakeFiles/aom_version.dir/build.make CMakeFiles/aom_version.dir/depend
#6 16.66 make[2]: Entering directory '/aom/build_tmp'
#6 16.66 cd /aom/build_tmp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /aom /aom /aom/build_tmp /aom/build_tmp /aom/build_tmp/CMakeFiles/aom_version.dir/DependInfo.cmake --color=
#6 16.68 make[2]: Leaving directory '/aom/build_tmp'
#6 16.68 make  -f CMakeFiles/aom_version.dir/build.make CMakeFiles/aom_version.dir/build
#6 16.68 make[2]: Entering directory '/aom/build_tmp'
#6 16.69 [  1%] Building C object CMakeFiles/aom_version.dir/gen_src/aom_version_dummy.c.o
#6 16.69 /usr/bin/cc  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c99 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wimplicit-function-declaration -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=100000 -Wshadow -Wundef -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -MD -MT CMakeFiles/aom_version.dir/gen_src/aom_version_dummy.c.o -MF CMakeFiles/aom_version.dir/gen_src/aom_version_dummy.c.o.d -o CMakeFiles/aom_version.dir/gen_src/aom_version_dummy.c.o -c /aom/build_tmp/gen_src/aom_version_dummy.c
#6 16.77 [  1%] Linking C static library libaom_version.a
#6 16.77 /usr/bin/cmake -P CMakeFiles/aom_version.dir/cmake_clean_target.cmake
#6 16.79 /usr/bin/cmake -E cmake_link_script CMakeFiles/aom_version.dir/link.txt --verbose=1
#6 16.80 /usr/bin/ar qc libaom_version.a CMakeFiles/aom_version.dir/gen_src/aom_version_dummy.c.o
#6 16.81 /usr/bin/ranlib libaom_version.a
#6 16.82 make[2]: Leaving directory '/aom/build_tmp'
#6 16.85 [  2%] Built target aom_version
#6 16.85 make  -f CMakeFiles/aom_pc.dir/build.make CMakeFiles/aom_pc.dir/depend
#6 16.85 make[2]: Entering directory '/aom/build_tmp'
#6 16.86 [  3%] Writing aom.pc
#6 16.86 /usr/bin/cmake -DAOM_CONFIG_DIR=/aom/build_tmp -DAOM_ROOT=/aom -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_INSTALL_BINDIR=bin -DCMAKE_INSTALL_INCLUDEDIR=include -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_PROJECT_NAME=AOM -DCONFIG_MULTITHREAD=1 -DCONFIG_TUNE_VMAF=0 -DCONFIG_TUNE_BUTTERAUGLI=0 -DHAVE_PTHREAD_H=1 -P /aom/build/cmake/pkg_config.cmake
#6 16.90 cd /aom/build_tmp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /aom /aom /aom/build_tmp /aom/build_tmp /aom/build_tmp/CMakeFiles/aom_pc.dir/DependInfo.cmake --color=
#6 16.90 make[2]: Leaving directory '/aom/build_tmp'
#6 16.92 make  -f CMakeFiles/aom_pc.dir/build.make CMakeFiles/aom_pc.dir/build
#6 16.93 make[2]: Entering directory '/aom/build_tmp'
#6 16.94 [  3%] Building C object CMakeFiles/aom_pc.dir/gen_src/aom_pc_dummy.c.o
#6 16.94 /usr/bin/cc  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c99 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wimplicit-function-declaration -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=100000 -Wshadow -Wundef -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -MD -MT CMakeFiles/aom_pc.dir/gen_src/aom_pc_dummy.c.o -MF CMakeFiles/aom_pc.dir/gen_src/aom_pc_dummy.c.o.d -o CMakeFiles/aom_pc.dir/gen_src/aom_pc_dummy.c.o -c /aom/build_tmp/gen_src/aom_pc_dummy.c
#6 17.02 [  3%] Linking C static library libaom_pc.a
#6 17.02 /usr/bin/cmake -P CMakeFiles/aom_pc.dir/cmake_clean_target.cmake
#6 17.03 /usr/bin/cmake -E cmake_link_script CMakeFiles/aom_pc.dir/link.txt --verbose=1
#6 17.06 /usr/bin/ar qc libaom_pc.a CMakeFiles/aom_pc.dir/gen_src/aom_pc_dummy.c.o
#6 17.06 /usr/bin/ranlib libaom_pc.a
#6 17.07 make[2]: Leaving directory '/aom/build_tmp'
#6 17.08 [  3%] Built target aom_pc
#6 17.08 make  -f CMakeFiles/aom_rtcd.dir/build.make CMakeFiles/aom_rtcd.dir/depend
#6 17.09 make[2]: Entering directory '/aom/build_tmp'
#6 17.09 cd /aom/build_tmp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /aom /aom /aom/build_tmp /aom/build_tmp /aom/build_tmp/CMakeFiles/aom_rtcd.dir/DependInfo.cmake --color=
#6 17.10 make[2]: Leaving directory '/aom/build_tmp'
#6 17.10 make  -f CMakeFiles/aom_rtcd.dir/build.make CMakeFiles/aom_rtcd.dir/build
#6 17.10 make[2]: Entering directory '/aom/build_tmp'
#6 17.11 [  3%] Building C object CMakeFiles/aom_rtcd.dir/aom_dsp/aom_dsp_rtcd.c.o
#6 17.11 /usr/bin/cc  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c99 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wimplicit-function-declaration -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=100000 -Wshadow -Wundef -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -MD -MT CMakeFiles/aom_rtcd.dir/aom_dsp/aom_dsp_rtcd.c.o -MF CMakeFiles/aom_rtcd.dir/aom_dsp/aom_dsp_rtcd.c.o.d -o CMakeFiles/aom_rtcd.dir/aom_dsp/aom_dsp_rtcd.c.o -c /aom/aom_dsp/aom_dsp_rtcd.c
#6 19.83 [  3%] Building CXX object CMakeFiles/yuv.dir/third_party/libyuv/source/cpu_id.cc.o
#6 19.85 /usr/bin/c++  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c++11 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=240000 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wno-unused-parameter -MD -MT CMakeFiles/yuv.dir/third_party/libyuv/source/cpu_id.cc.o -MF CMakeFiles/yuv.dir/third_party/libyuv/source/cpu_id.cc.o.d -o CMakeFiles/yuv.dir/third_party/libyuv/source/cpu_id.cc.o -c /aom/third_party/libyuv/source/cpu_id.cc
#6 20.11 [  3%] Building CXX object CMakeFiles/yuv.dir/third_party/libyuv/source/planar_functions.cc.o
#6 20.11 /usr/bin/c++  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c++11 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=240000 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wno-unused-parameter -MD -MT CMakeFiles/yuv.dir/third_party/libyuv/source/planar_functions.cc.o -MF CMakeFiles/yuv.dir/third_party/libyuv/source/planar_functions.cc.o.d -o CMakeFiles/yuv.dir/third_party/libyuv/source/planar_functions.cc.o -c /aom/third_party/libyuv/source/planar_functions.cc
#6 21.40 [  3%] Building C object CMakeFiles/aom_rtcd.dir/aom_scale/aom_scale_rtcd.c.o
#6 21.40 /usr/bin/cc  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c99 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wimplicit-function-declaration -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=100000 -Wshadow -Wundef -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -MD -MT CMakeFiles/aom_rtcd.dir/aom_scale/aom_scale_rtcd.c.o -MF CMakeFiles/aom_rtcd.dir/aom_scale/aom_scale_rtcd.c.o.d -o CMakeFiles/aom_rtcd.dir/aom_scale/aom_scale_rtcd.c.o -c /aom/aom_scale/aom_scale_rtcd.c
#6 21.50 [  4%] Building C object CMakeFiles/aom_rtcd.dir/av1/common/av1_rtcd.c.o
#6 21.51 /usr/bin/cc  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c99 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wimplicit-function-declaration -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=100000 -Wshadow -Wundef -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -MD -MT CMakeFiles/aom_rtcd.dir/av1/common/av1_rtcd.c.o -MF CMakeFiles/aom_rtcd.dir/av1/common/av1_rtcd.c.o.d -o CMakeFiles/aom_rtcd.dir/av1/common/av1_rtcd.c.o -c /aom/av1/common/av1_rtcd.c
#6 22.33 make[2]: Leaving directory '/aom/build_tmp'
#6 22.35 [  5%] Built target aom_rtcd
#6 22.36 make  -f CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/build.make CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/depend
#6 22.36 make[2]: Entering directory '/aom/build_tmp'
#6 22.36 cd /aom/build_tmp && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /aom /aom /aom/build_tmp /aom/build_tmp /aom/build_tmp/CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/DependInfo.cmake --color=
#6 22.37 make[2]: Leaving directory '/aom/build_tmp'
#6 22.37 make  -f CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/build.make CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/build
#6 22.37 make[2]: Entering directory '/aom/build_tmp'
#6 22.41 [  5%] Building C object CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_quantize_avx2.c.o
#6 22.41 /usr/bin/cc  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c99 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wimplicit-function-declaration -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=100000 -Wshadow -Wundef -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -mavx2 -mno-avx256-split-unaligned-load -mno-avx256-split-unaligned-store -MD -MT CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_quantize_avx2.c.o -MF CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_quantize_avx2.c.o.d -o CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_quantize_avx2.c.o -c /aom/av1/encoder/x86/av1_quantize_avx2.c
#6 23.40 [  5%] Building CXX object CMakeFiles/yuv.dir/third_party/libyuv/source/row_any.cc.o
#6 23.40 /usr/bin/c++  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c++11 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=240000 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wno-unused-parameter -MD -MT CMakeFiles/yuv.dir/third_party/libyuv/source/row_any.cc.o -MF CMakeFiles/yuv.dir/third_party/libyuv/source/row_any.cc.o.d -o CMakeFiles/yuv.dir/third_party/libyuv/source/row_any.cc.o -c /aom/third_party/libyuv/source/row_any.cc
#6 24.10 [  5%] Building C object CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_highbd_quantize_avx2.c.o
#6 24.10 /usr/bin/cc  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c99 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wimplicit-function-declaration -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=100000 -Wshadow -Wundef -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -mavx2 -mno-avx256-split-unaligned-load -mno-avx256-split-unaligned-store -MD -MT CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_highbd_quantize_avx2.c.o -MF CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_highbd_quantize_avx2.c.o.d -o CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/av1_highbd_quantize_avx2.c.o -c /aom/av1/encoder/x86/av1_highbd_quantize_avx2.c
#6 25.67 [  6%] Building C object CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/corner_match_avx2.c.o
#6 25.67 /usr/bin/cc  -I/aom -I/aom/build_tmp -I/aom/apps -I/aom/common -I/aom/examples -I/aom/stats -I/aom/third_party/libyuv/include -O3 -s -static-libgcc -fno-strict-overflow -fstack-protector-all -fPIC -DNDEBUG -O3 -DNDEBUG -std=c99 -Wall -Wdisabled-optimization -Wextra -Wfloat-conversion -Wformat=2 -Wimplicit-function-declaration -Wlogical-op -Wpointer-arith -Wsign-compare -Wtype-limits -Wuninitialized -Wunused -Wvla -Wstack-usage=100000 -Wshadow -Wundef -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -mavx2 -mno-avx256-split-unaligned-load -mno-avx256-split-unaligned-store -MD -MT CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/corner_match_avx2.c.o -MF CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/corner_match_avx2.c.o.d -o CMakeFiles/aom_av1_encoder_avx2_intrinsics.dir/av1/encoder/x86/corner_match_avx2.c.o -c /aom/av1/encoder/x86/corner_match_avx2.c
```

**Comment by wader** [comment]
Think it looks good expect see if the rpath thing make a difference for static PIE builds (think confusing to keep it if not) and not sure we should change NDEBUG,  i think it mostly controls if asserts are included or not? by not setting it i guess it will be more up to each libs? personally in my experience debugging crashes i think asserts are worth keeping even in release builds.

**Comment by mathieu-aubin** [comment]
> Think it looks good expect see if the rpath thing make a difference for static PIE builds (think confusing to keep it if not)

as per [Wikipedia](https://en.wikipedia.org/wiki/Rpath)
 ... 

"The rpath of an executable or shared library is an optional entry in the .dynamic section of the [ELF](https://en.wikipedia.org/wiki/Executable_and_Linkable_Format) executable or shared libraries, with the type DT_RPATH, called the DT_RPATH attribute. It can be stored there at link time by the linker."

and

"Static libraries are searched and linked into the ELF file at link time and are not searched at run time."

Meaning any *rpath* becomes irrelevant information that most likely will bloat filea without the need to, in my understanding

> not sure we should change NDEBUG, i think it mostly controls if asserts are included or not? by not setting it i guess it will be more up to each libs? personally in my experience debugging crashes i think asserts are worth keeping even in release builds.

Im not sure but i am of the strong opinion that `-DNDEBUG` is used mainly on releases type builds in order to shrink library/executable sizes by not adding debug asserts.

While you might want to keep them, they are mainly useless in a static build and more than likely, already are being removed from using such optimization flags as `-O3`.

In cmake, per example, `CMAKE_BUILD_TYPE=Release` generally includes it (minsizerel too i believe) by default unless tricked out intentionally. Explicit use is my preference over anything else (especially when this trivial) but i understand its not shared by everyone.

**Comment by wader** [comment]
> Meaning any _rpath_ becomes irrelevant information that most likely will bloat filea without the need to, in my understanding

Yes, sorry i should have been clearer. What i meant was that i suspected that when linking as static PIE there wont be any need for a runtime path so is probably not included. Now when i read about it i think it's actually usually not even included in dynamic ELFs as there are default paths that are searched (ld.conf?). Got a bit curious and poked about with objdump in a static-ffmpeg binary

```
$ objdump -x ffmpeg

ffmpeg:	file format elf64-x86-64
architecture: x86_64
start address: 0x0000000000330223

...

Dynamic Section:
  INIT         0x000000000024c000
  FINI         0x0000000003a2dab1
  INIT_ARRAY   0x00000000046ebf10
  INIT_ARRAYSZ 0x00000000000001a8
  FINI_ARRAY   0x00000000046ec0b8
  FINI_ARRAYSZ 0x0000000000000020
  GNU_HASH     0x00000000000002c8
  STRTAB       0x0000000000000300
  SYMTAB       0x00000000000002e8
  STRSZ        0x0000000000000001
  SYMENT       0x0000000000000018
  DEBUG        0x0000000000000000
  PLTGOT       0x00000000048d3bd0
  RELA         0x0000000000000308
  RELASZ       0x000000000024b550
  RELAENT      0x0000000000000018
  BIND_NOW     0x0000000000000000
  FLAGS_1      0x0000000008000001
  RELACOUNT    0x000000000001878e

...
```

> Im not sure but i am of the strong opinion that `-DNDEBUG` is used mainly on releases type builds in order to shrink library/executable sizes by not adding debug asserts.

I would guess the size difference is usually minimal? think i'm would be more concerned about runtime performance of doing assert checks.

> While you might want to keep them, they are mainly useless in a static build and more than likely, already are being removed from using such optimization flags as `-O3`.

Ok but at least `assert.h`-asserts i don't think will get stripped out as they might have the side effect of print to stderr and abort.

> In cmake, per example, `CMAKE_BUILD_TYPE=Release` generally includes it (minsizerel too i believe) by default unless tricked out intentionally. Explicit use is my preference over anything else (especially when this trivial) but i understand its not shared by everyone.

I think i prefer to let each library decide how asserts are affected by build type. Some probably prefer to not include them (maybe because they are slow and do extensive validations) and other library always include them because they are fast and will help a great deal debug cashes.

As the rpath thing seems to not make difference and we can probably continue the NDEBUG talk it would be great if this PR was just about doing cosmetic maybe?

**Comment by mathieu-aubin** [comment]
should have rpath and ndebug out of the picture now

**Comment by wader** [comment]
thanks! will merge once build is green

**Pull request merged**